### DISCUSSION TITLE
Can i see the otlp exporter request status (.net)?

### DISCUSSION THREAD

**Discussion by Wholteza** [open-telemetry/opentelemetry-dotnet#5379] [Q&A]
I'm building an application which acts as a log proxy for event driven communication.
It's a .net 6 azure function app that receives events and logs information about them to elastic.co though the open telemetry .net sdk + the otlp exporter.

I want to be 100% sure that the application has sent a log message for each and every event received.

The recommended way to use the logger seems to be to register the OpenTelemetryLoggerProvider as an implementation of the ILoggerProvider interface. You then resolve the ILoggerProvider through the IOC container and create a logger instance, or you just resolve an ILogger<YourClass> directly.

However the ILogger implementations are meant to be designed in a safe way that swallows possible exceptions.
Unfortunately that makes it seemingly impossible for me to know if the message was successfully logged or not.

Is there a way to get information about the otlp exporters status either prior to or after logging so that i can choose to mark the current event as unhandled and wait for a successful connection?

**Comment by cijothomas** [comment]
A solution like https://github.com/open-telemetry/opentelemetry-dotnet/issues/3880 will help get overall stats/health of the sdk/exporters etc. It is not unreasonable to add some capability to get notified when Exports face exceptions, but no work is planned in that regard. 

The exporters(and every component) reports its internal error via EventSource, so you can subscribe to it, to get notified when Exporter is dropping items. (If used with BatchExporter, you'll know that a batch failed, but won't be able to tell which particular records were part of it.)

**Reply by Wholteza** [reply]
That might work.
Are there any examples of how to subscribe to these events ? :)
